src/Tvheadend.cpp

```diff
@@ -1385,13 +1385,9 @@ void CTvheadend::ParseRecordingAddOrUpdate ( htsmsg_t *msg, bool bAdd )
   {
     UPDATE(rec.state, PVR_TIMER_STATE_COMPLETED);
   }
-  else if (strstr(state, "missed") != NULL)
-  {
-    UPDATE(rec.state, PVR_TIMER_STATE_ERROR);
-  }
-  else if (strstr(state, "invalid")   != NULL)
+  else
   {
-    UPDATE(rec.state, PVR_TIMER_STATE_ERROR);
+    UPDATE(rec.error, state);
```

Comment:
No, we can't use "PVR_TIMER_STATE_ERROR" for invalid or missed recordings, if we do, these recordings will show up between the timers with "IsTimer()" and those are no timers. 

Not setting a state for missing and invalid recordings will leave them as "PVR_TIMER_STATE_NEW"
Everything with "PVR_TIMER_STATE_NEW" won't be transferred to kodi as it's not in IsTimer() or IsRecording(), this is how it should be as we don't want them in kodi. 
I needed to change this to keep the old behavior after I added "PVR_TIMER_STATE_ERROR" to IsTimer().